lightning/src/offers/invoice_request.rs

```diff
@@ -335,52 +338,75 @@ impl<'a, 'b, P: PayerIdStrategy, T: secp256k1::Signing> InvoiceRequestBuilder<'a
 		self
 	}
 
-	pub(super) fn build_unchecked(self) -> UnsignedInvoiceRequest<'a> {
+	pub(super) fn build_unchecked(self) -> UnsignedInvoiceRequest {
 		self.build_without_checks().0
 	}
 }
 
 /// A semantically valid [`InvoiceRequest`] that hasn't been signed.
-pub struct UnsignedInvoiceRequest<'a> {
-	offer: &'a Offer,
+pub struct UnsignedInvoiceRequest {
+	bytes: Vec<u8>,
 	invoice_request: InvoiceRequestContents,
+	tagged_hash: TaggedHash,
 }
 
-impl<'a> UnsignedInvoiceRequest<'a> {
-	/// Signs the invoice request using the given function.
-	///
-	/// This is not exported to bindings users as functions are not yet mapped.
-	pub fn sign<F, E>(self, sign: F) -> Result<InvoiceRequest, SignError<E>>
-	where
-		F: FnOnce(&Message) -> Result<Signature, E>
-	{
+impl UnsignedInvoiceRequest {
```

Comment:
Yeah, was thinking individual methods for accessing parts of the content of interest. Could expose the entire contents itself, but I was hoping to avoid that.

Reply:
here and elsewhere (e.g. `UnsignedBolt12Invoice`), a validating signer needs to be able to go from the struct to the underlying bytes and back, because it may need to send the data to a remote signer.

Reply:
Added parsing and serialization. Which fields will you need access to? I'm considering how best to expose them.

Reply:
pretty much all fields for invoices, except for the routing stuff.  but as I mentioned somewhere else in this PR, validating signers might not be that interested in offers or invoice requests, because you can't lose money with bad ones, only with bad invoices.

Reply:
ah, given the discussion in https://github.com/lightningdevkit/rust-lightning/pull/2432#discussion_r1292195453, I think we do care about offers and invoice requests.  we also need to cryptographically check the linkage between the different object types.

Reply:
(e.g. check that an invoice's embedded offer matches a previously seen offer)